src/utilities/map.ts

```diff
@@ -34,3 +34,30 @@ export function getTileElement(x: number, y: number, elementIdx: number): TileEl
 	const tile = getTileByCoords(x, y);
 	return tile.getElement(elementIdx);
 }
+
+/**
+ * Finds the track element index on a tile that matches the given track location (car.trackLocation).
+ */
+export function findTrackElementIndex(trackLocation: CarTrackLocation): number | null 
```

Comment:
There's a similar function in `spacingEditor.ts` already. Maybe that one can be moved here and reused, and add `trackType` as an optional second parameter?